File: leetcode/58-length-of-last-word/solution.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        wordStart = False
        count = 0
        i = len(s) - 1
        while i >= 0:
            if s[i] == ' ':
                if wordStart:
                    break
                i -= 1
                continue
            count += 1
            wordStart = True
            i -= 1
        return count
        

File: leetcode/58-length-of-last-word/Solution.java
class Solution {
    public int lengthOfLastWord(String s) {
        int count = 0;
        boolean wordStart = false;
        for (int i = s.length() - 1; i >= 0; i--) {
            if (s.charAt(i) == ' ') {
                if (wordStart) break;
                continue;
            }
            wordStart = true;
            count++;
        }
        return count;
    }
}

